src/App.tsx

```diff
@@ -15,35 +16,122 @@
+
+  switch (sortingType) {
+    case SortType.ALPHABETICALLY:
+      goodsArrayCopy.sort((good1, good2) => good1.localeCompare(good2));
+      break;
+
+    case SortType.LENGTH:
+      goodsArrayCopy.sort((good1, good2) => good1.length - good2.length);
+      break;
+
+    default:
+      break;
+  }
+
+  if (isReversed === true) {
+    return goodsArrayCopy.reverse();
+  } else {
+    return goodsArrayCopy;
+  }
+};
+
 export const App: React.FC = () => {
+  const [sortingType, setSortingType] = useState<SortType>(SortType.DEFAULT);
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+
+  const handleReset = () => {
+    setSortingType(SortType.DEFAULT);
+    setIsReversed(false);
+  };
+
+  const handleReverse = () => {
+    if (isReversed === false) {
+      setIsReversed(true);
+    }
+
+    if (isReversed === true) {
+      setIsReversed(false);
+    }
+  };
+
+  const sortedGoods = sortingGoodsFunction(goodsFromServer, {
+    sortingType,
+    isReversed,
+  });
+
+  const isNoSorting = sortingType === SortType.DEFAULT && isReversed === false;
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          onClick={() => setSortingType(SortType.ALPHABETICALLY)}
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sortingType !== SortType.ALPHABETICALLY,
+          })}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          onClick={() => setSortingType(SortType.LENGTH)}
+          type="button"
+          className={cn('button is-success', {
+            'is-light': sortingType !== SortType.LENGTH,
+          })}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          onClick={handleReverse}
+          type="button"
+          className={cn('button is-warning', {
+            'is-light': !isReversed,
+          })}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {!isNoSorting ? (
+          <button
+            onClick={handleReset}
+            type="button"
+            className="button is-danger is-light"
+          >
+            Reset
+          </button>
+        ) : (
+          ''
+        )}
       </div>
 
       <ul>
         <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
+          {sortedGoods.map(good => (
+            <li key={good} data-cy="Good">
+              {good}
+            </li>
+          ))}
```

Comment:
[CODE KNOWLEDGE] - The `key` prop in the list items should be a unique identifier for each item. Using `good` as the key might lead to issues if the list contains duplicate values. Consider using an index or another unique value as the key.